Userland/Libraries/LibJS/Runtime/Object.h

```diff
@@ -150,18 +150,30 @@ class Object : public Cell {
     IndexedProperties& indexed_properties() { return m_indexed_properties; }
     void set_indexed_property_elements(Vector<Value>&& values) { m_indexed_properties = IndexedProperties(move(values)); }
 
-    [[nodiscard]] Value invoke_internal(PropertyName const&, Optional<MarkedValueList> arguments);
+    [[nodiscard]] Value invoke_internal(Value this_value, PropertyName const&, Optional<MarkedValueList> arguments);
 
     template<typename... Args>
     [[nodiscard]] ALWAYS_INLINE Value invoke(PropertyName const& property_name, Args... args)
     {
         if constexpr (sizeof...(Args) > 0) {
             MarkedValueList arglist { heap() };
             (..., arglist.append(move(args)));
-            return invoke(property_name, move(arglist));
+            return invoke(this, property_name, move(arglist));
         }
 
-        return invoke(property_name);
+        return invoke(this, property_name);
+    }
+
+    template<typename... Args>
+    [[nodiscard]] ALWAYS_INLINE Value invoke(Value this_value, PropertyName const& property_name, Args... args)
```

Comment:
I did not add a spec link here since it is not completely the same and we now have to functions as well which makes it even more complicated.